ui/components/NodeInputEmail.tsx

```diff
@@ -23,6 +23,13 @@ export const NodeInputEmail: FC<NodeInputProps> = ({
 
   const error = hasLocalValidation ? localError : upstreamError;
 
+  const submitBtn = document.getElementsByClassName("p-button--positive")?.[0];
+  if (error) {
+    submitBtn?.setAttribute("disabled", "");
+  } else {
+    submitBtn?.removeAttribute("disabled");
+  }
+
```

Comment:
best practice here would be to do dom manipulation (like these attribute changes) inside a `useEffect` function

Reply:
Added the use effect. Actually, not sure if this is an improvement, because it is adding more code and complexity for a very tiny micro optimization.